lte/protos/subscriberdb.proto

```diff
@@ -213,25 +219,62 @@ message SubscribersDigest {
   string md5_base64_digest = 1;
 }
 
+message CheckSubscribersInSyncRequest {
+  // flat_digest is the deterministic digest of the full set of subscriber
+  // objects stored on the client side.
+  SubscribersDigest flat_digest = 1;
+}
+
+message CheckSubscribersInSyncResponse {
+  // in_sync is true if client's existing subscribers match those on the server.
+  bool in_sync = 1;
+}
+
+message SyncSubscribersRequest {
+  // per_sub_digest is a set of digests for each client-side subscriber, indexed
+  // by their IDs (IMSI).
+  map<string, SubscribersDigest> per_sub_digests = 1;
+}
+
+message SyncSubscribersResponse {
+  // resync is true if the client-server data difference is too big and an
+  // overall resync is needed. If true, all subscriber changesets will be empty.
+  bool resync = 1;
+  // flat_digest is the deterministic digest of all server-side subscribers.
+  SubscribersDigest flat_digest = 2;
+  // per_sub_digests is a set of digests for each server-side subscriber, indexed
+  // by their IDs (IMSI).
+  map<string, SubscribersDigest> per_sub_digests = 3;
+  // added_modified is the set of all subscriber data modified/added on the server
+  // side, indexed by their IDs (IMSI).
+  map<string, SubscriberDataWithDigest> added_modified = 4;
```

Comment:
I'd still prefer to separate out the subs from the digests, using two separate fields -- no need for the `SubscriberDataWithDigest` type